### ISSUE TITLE
2.2.0 writes negative timestamps

### ISSUE THREAD

**Issue by grassick** [gulpjs/vinyl-fs#113]
2.2.0 writes negative timestamps under some circumstances:

```
-rw-r--r--  1 clayton clayton 1238 Oct 15 11:49 index_map.html
-rw-r--r--  1 clayton clayton 1192 -9223372036854775808 manifest.appcache
-rw-r--r--  1 clayton clayton  196 Feb 20  2015 Readme.md
-rw-r--r--  1 clayton clayton   25 Feb 20  2015 robots.txt
```

2.1.1 did not. I see recent commits with mtime changes. This is in node 0.12.7.

**Comment by grassick** [comment]
2.2.0 also breaks watchify by causing `end` events to not fire anymore in gulp. Still investigating that problem.

**Comment by phated** [comment]
Can you give me a reproduction case? We added a new feature in 2.2.0 to set mtime and atime if set on a vinyl object.

**Comment by phated** [comment]
cc'ing @piranna to look into this also

**Comment by piranna** [comment]
Maybe the original file has badly set the mtime of atime fields? Since now we are using them, maybe it have shown a hidden bug elsewhere...

**Comment by phated** [comment]
@piranna I wonder if this could come up because node 0.12 added properties to the fs.Stats object and made the atime, mtime, etc Date objects and passing undefined to `new Date()` causes an invalid date.  Thoughts?

**Comment by phated** [comment]
@piranna maybe we need a check like http://stackoverflow.com/a/10589791 inside the utimes helper?

**Comment by piranna** [comment]
That time converted to hexadecimal is `FFFFFFFFFA80A008`... A little bit big, isn't it? I'm not sure if it's a special value or something...


> @piranna maybe we need a check like http://stackoverflow.com/a/10589791 inside the utimes helper?

I would not hurt in any case... What should do in case of a bad value, ignore it or raise an error?

**Comment by phated** [comment]
@piranna due to node 0.12 having NaN dates in my quick tests (run `new fs.Stats()` in the repl), I think we have to just ignore them.

**Comment by piranna** [comment]
I've just done a [pull-request](https://github.com/gulpjs/vinyl-fs/issues/114) to fix this. It ignore the atime and mtime fields if any of both is invalid.

**Comment by phated** [comment]
Sorry for the delay.  I wanted to make sure our tests had a reproduction case and I could verify we fixed this before was published.  I've just published this as 2.2.1 - please let me know how this works out for you.
